src/main/java/io/neonbee/internal/processor/EntityProcessor.java

```diff
@@ -90,90 +92,99 @@ public void init(OData odata, ServiceMetadata serviceMetadata) {
+             * return=representation, or by specifying the system query options $select or $expand.
+             */
+            response.setStatusCode(HttpStatusCode.NO_CONTENT.getStatusCode());
+            processPromise.complete();
+        });
     }
 
     @Override
     public void deleteEntity(ODataRequest request, ODataResponse response, UriInfo uriInfo)
             throws ODataApplicationException, ODataLibraryException {
-        /*
-         * See https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_URLComponents
-         * for details about how the OData URL broken down into its component parts.
-         */
-        UriResourceEntitySet uriResourceEntitySet = (UriResourceEntitySet) uriInfo.getUriResourceParts().get(0);
-        handleEntityIgnoringBody(uriResourceEntitySet.getEntityType(), request, DELETE,
-                handleDeleteEntityResult(response));
-    }
+        Promise<Void> processPromise = getProcessPromise();
 
-    private void handleEntityIgnoringBody(EdmEntityType edmEntityType, ODataRequest request, DataAction action,
-            Handler<AsyncResult<EntityWrapper>> entityHandler) {
-        handleEntity(edmEntityType, request, action, null, entityHandler);
+        UriResourceEntitySet uriResourceEntitySet = (UriResourceEntitySet) uriInfo.getUriResourceParts().get(0);
+        forwardRequest(request, uriResourceEntitySet, null, DELETE, processPromise).onSuccess(ew -> {
+            response.setStatusCode(HttpStatusCode.NO_CONTENT.getStatusCode());
+            processPromise.complete();
+        });
     }
 
-    private void handleEntity(UriResourceEntitySet uriResourceEntitySet, ODataRequest request,
-            ContentType requestFormat, DataAction action, Handler<AsyncResult<EntityWrapper>> entityHandler)
+    private Entity parseBody(ODataRequest request, UriResourceEntitySet uriResourceEntitySet, ContentType requestFormat)
             throws DeserializerException {
-        EdmEntityType edmEntityType = uriResourceEntitySet.getEntitySet().getEntityType();
-        Entity entity = odata.createDeserializer(requestFormat).entity(request.getBody(), edmEntityType).getEntity();
-        if (UPDATE.equals(action)) {
-            EdmHelper.addKeyPredicateValues(entity, uriResourceEntitySet, routingContext);
-        }
-        handleEntity(edmEntityType, request, action, entity, entityHandler);
+        EdmEntityType entityType = uriResourceEntitySet.getEntitySet().getEntityType();
+        return odata.createDeserializer(requestFormat).entity(request.getBody(), entityType).getEntity();
     }
 
-    void handleEntity(EdmEntityType edmEntityType, ODataRequest request, DataAction action, Entity entity,
-            Handler<AsyncResult<EntityWrapper>> entityHandler) {
+    @VisibleForTesting
+    Future<EntityWrapper> forwardRequest(ODataRequest request, UriResourceEntitySet uriResourceEntitySet, Entity entity,
```

Comment:
i guess this method can be declared as private and remove visible for testing pls